test/e2e/providers/vsphere/vsphere.go

```diff
@@ -133,9 +140,271 @@ func (p *Provider) GetType() config.PlatformType {
+	// https://docs.openshift.com/container-platform/4.12/storage/dynamic-provisioning.html#about_dynamic-provisioning
+	sc, err := p.ensureStorageClass(client)
+	if err != nil {
+		return nil, fmt.Errorf("unable to ensure a usable StorageClass is created: %w", err)
+	}
+	pvcSpec := core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			GenerateName: storageClassName + "-",
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			AccessModes: []core.PersistentVolumeAccessMode{core.ReadWriteOnce},
+			Resources: core.ResourceRequirements{
+				Requests: core.ResourceList{core.ResourceStorage: resource.MustParse("2Gi")},
+			},
+			StorageClassName: &sc.Name,
+		},
+	}
+	return client.CoreV1().PersistentVolumeClaims(namespace).Create(context.TODO(), &pvcSpec, meta.CreateOptions{})
+}
+
+// ensureStorageClass ensures a usable vSphere NTFS storage class exists
+func (p *Provider) ensureStorageClass(client client.Interface) (*storage.StorageClass, error) {
+	sc, err := client.StorageV1().StorageClasses().Get(context.TODO(), storageClassName, meta.GetOptions{})
+	if err == nil {
+		return sc, nil
+	} else if !k8sapierrors.IsNotFound(err) {
+		return nil, fmt.Errorf("error getting storage class '%s': %w", storageClassName, err)
+	}
+	volumeBinding := storage.VolumeBindingImmediate
+	reclaimPolicy := core.PersistentVolumeReclaimDelete
+	sc = &storage.StorageClass{
+		ObjectMeta: meta.ObjectMeta{
+			Name: storageClassName,
+		},
+		Provisioner:       "csi.vsphere.vmware.com",
+		Parameters:        map[string]string{"fstype": "ntfs"},
+		ReclaimPolicy:     &reclaimPolicy,
+		VolumeBindingMode: &volumeBinding,
+	}
+	return client.StorageV1().StorageClasses().Create(context.TODO(), sc, meta.CreateOptions{})
 }
 
-func (p *Provider) CreatePVC(_ client.Interface, _ string) (*core.PersistentVolumeClaim, error) {
-	return nil, fmt.Errorf("storage not supported on vSphere")
+// ensureWindowsCSIDrivers ensures that the vSphere CSI drivers are deployed across Windows nodes
+func (p *Provider) ensureWindowsCSIDrivers(client client.Interface) error {
+	csiNamespace := "openshift-cluster-csi-drivers"
+	// The FSS used by Linux nodes is unusable as it does not set csi-windows-support to true.
+	windowsFSS := "win-internal-feature-states.csi.vsphere.vmware.com"
+	fssCM := core.ConfigMap{
+		ObjectMeta: meta.ObjectMeta{
+			Name: windowsFSS,
+		},
+		Data: map[string]string{
+			"block-volume-snapshot":            "true",
+			"cnsmgr-suspend-create-volume":     "true",
+			"csi-auth-check":                   "true",
+			"csi-migration":                    "true",
+			"improved-csi-idempotency":         "true",
+			"improved-volume-topology":         "false",
+			"online-volume-extend":             "true",
+			"topology-preferential-datastores": "true",
+			"csi-windows-support":              "true",
+		},
+		BinaryData: nil,
+	}
+	_, err := client.CoreV1().ConfigMaps(csiNamespace).Create(context.TODO(), &fssCM, meta.CreateOptions{})
```

Comment:
Would this become a problem whenever we have the real upgrade job in CI? i.e. should we be ensuring it exists rather than always creating it.